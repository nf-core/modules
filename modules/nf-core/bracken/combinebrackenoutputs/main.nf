process BRACKEN_COMBINEBRACKENOUTPUTS {
    tag "$meta.id"
    label 'process_low'

    conda "bioconda::bracken=2.7"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/bracken:2.7--py39hc16433a_0':
        'quay.io/biocontainers/bracken:2.7--py39hc16433a_0' }"

    input:
    tuple val(meta), path(input)

    output:
    tuple val(meta), path("*.txt"), emit: txt
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    // WARN: Version information not provided by tool on CLI.
    // Please update version string below when bumping container versions.
    def VERSION = '2.7'
    """
    combine_bracken_outputs.py \\
        $args \\
        --files ${input} \\
        -o ${prefix}.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        combine_bracken_output: ${VERSION}
    END_VERSIONS
    """
}
