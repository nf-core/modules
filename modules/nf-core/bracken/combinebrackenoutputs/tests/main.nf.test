nextflow_process {

    name "Test Process BRACKEN_COMBINEBRACKENOUTPUTS"
    script "../main.nf"
    process "BRACKEN_COMBINEBRACKENOUTPUTS"

    tag "modules"
    tag "modules_nfcore"
    tag "bracken"
    tag "bracken/combinebrackenoutputs"
    tag "bracken/bracken"
    tag "kraken2/kraken2"
    tag "untar"

    setup {
        run ("UNTAR") {
            script "../../../untar/main.nf"
            process {
                """
                input[0] = [[],file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/db/kraken2_bracken.tar.gz', checkIfExists: true)]
                """
            }
        }

        run("KRAKEN2_KRAKEN2") {
            script "../../../kraken2/kraken2/main.nf"
            process {
                """
                input[0] = Channel.of(
                                [
                                    [id: 'test', single_end: true], file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true),
                                    [id: 'test2', single_end: true], file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_2.fastq.gz', checkIfExists: true),
                                ]
                            )
                input[1] = UNTAR.out.untar.map{it[1]}
                input[2] = false
                input[3] = false
                """
            }
        }

        run("BRACKEN_BRACKEN") {
            script "../../bracken/main.nf"
            process {
                """
                input[0] = KRAKEN2_KRAKEN2.out.report
                input[1] = UNTAR.out.untar.map{it[1]}
                """
            }
        }

    }

    test("sarscov2 - fastq - fastq") {

        setup {

        }

        when {
            process {
                """
                input[0] = BRACKEN_BRACKEN.out.reports
                                            .map{it[1]}
                                            .collect()
                                            .map{ [ [id: 'db'], it ] }
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }
    test("sarscov2 - fastq - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = BRACKEN_BRACKEN.out.reports
                                            .map{it[1]}
                                            .collect()
                                            .map{ [ [id: 'db'], it ] }
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
