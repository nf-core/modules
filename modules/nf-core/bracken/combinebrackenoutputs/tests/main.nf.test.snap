{
    "sarscov2 - fastq - fastq": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "db"
                        },
                        "db.txt:md5,fb22ad13f9685d5bf5026542b9b2a6b8"
                    ]
                ],
                "1": [
                    "versions.yml:md5,8edd746a7a248643d2cbf1b7e576fe38"
                ],
                "txt": [
                    [
                        {
                            "id": "db"
                        },
                        "db.txt:md5,fb22ad13f9685d5bf5026542b9b2a6b8"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,8edd746a7a248643d2cbf1b7e576fe38"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.03.0"
        },
        "timestamp": "2024-04-16T15:11:43.6682881"
    },
    "sarscov2 - fastq - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "db"
                        },
                        "db.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,8edd746a7a248643d2cbf1b7e576fe38"
                ],
                "txt": [
                    [
                        {
                            "id": "db"
                        },
                        "db.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,8edd746a7a248643d2cbf1b7e576fe38"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.03.0"
        },
        "timestamp": "2024-04-16T15:11:50.686596"
    }
}